server.js

```diff
@@ -58,19 +62,60 @@ MongoClient.connect('mongodb://localhost:4201/tutorial-builder', (err, database)
 SERVER Routes
 
 */
+app.put('/changepass', (req, res) => {
+	db.collection('users').findOne({username: req.body.username}).then(user => {
+		if (!user) {
+			return res.send(boom.unauthorized('invalid username'));
+		}
+
+		return bcrypt.compare(req.body.password, user.password, function(err, resp) {
+			if(resp) {
+			  	bcrypt.hash(req.body.newPass, saltRounds, function(err, hash) {
+					if(hash) {
+						const updatedUser = {
+							username: user.username,
+							password: hash,
+							tutorialsUsed: user.tutorialsUsed,
+							tutorialsOwned: user.tutorialsOwned
+						}
+
+						db.collection("users").findOneAndUpdate(
+							{username: req.body.username},
```

Comment:
can use _id for db index efficiency benefits. 

`findOneAnd Update({_id: new ObjectID(req.user.id)}, ... `